src/Extension/ValueResolver/PriorityPathValueResolver.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/**
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerSdk\Sdk\Extension\ValueResolver;
+
+use SprykerSdk\Sdk\Core\Application\ValueResolver\ConfigurableAbstractValueResolver;
+use SprykerSdk\Sdk\Extension\Exception\UnresolvableValueExceptionException;
+use SprykerSdk\Sdk\Extension\ValueResolver\Enum\Type;
+use SprykerSdk\SdkContracts\Entity\ContextInterface;
+
+class PriorityPathValueResolver extends ConfigurableAbstractValueResolver
+{
+    /**
+     * {@inheritDoc}
+     *
+     * @return string
+     */
+    public function getId(): string
+    {
+        return 'PRIORITY_PATH';
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param \SprykerSdk\SdkContracts\Entity\ContextInterface $context
+     * @param array $settingValues
+     * @param bool $optional
+     *
+     * @throws \SprykerSdk\Sdk\Extension\Exception\UnresolvableValueExceptionException
+     *
+     * @return string
+     */
+    public function getValue(ContextInterface $context, array $settingValues, bool $optional = false): string
+    {
+        $relativePath = (string)parent::getValue($context, $settingValues, $optional);
+
+        foreach ($this->getSettingPaths() as $settingKey) {
+            $path = $settingValues[$settingKey];
+            if (strpos($path, DIRECTORY_SEPARATOR, -1) === 0) {
+                $path = rtrim($path, DIRECTORY_SEPARATOR);
+            }
+            $path = implode(DIRECTORY_SEPARATOR, [$path, $relativePath]);
+            if (file_exists($path)) {
+                return $this->formatValue($path);
+            }
+        }
+        if (!$this->getSettingPaths()) {
```

Comment:
It is more logical to place it before the `foreach`. Also, would be better to keep it in a separate method with a readable name, e.g. `createValuePathFromRelativePath`, which allows you to reuse it in the foreach loop above.